### DISCUSSION TITLE
Publishing docker image on release

### DISCUSSION THREAD

**Discussion by akash-akya** [libvips/libvips#3002] [Ideas]
Hi all, 
Thanks for creating and maintaining libvips!

Just wanted to know if you are open to publishing docker image for libvips as part of CI/CD setup.
Docker image will be useful for trying out a specific version irrespective of OS, running multiple versions, and simplifying CI/CD setup for the libvips based libraries. 

We can automate image creation on release, using GitHub Actions. If you are okay with it, I can open a PR to set it up. 

I tried it [here](https://github.com/akash-akya/vix/runs/7838534928?check_suite_focus=true). And it seems to be working fine. I used Dockerfile from https://github.com/marcbachmann/dockerfile-libvips

**Comment by jcupitt** [comment]
Hi @akash-akya,

We've talked about it (it was one of the reasons we moved to github actions earlier this year) but not got around to it.

Sure, a PR would be great, thank you. We'd probably need a small-and-safe build, a full-fat build, nightly, stable, and maybe one RPM-based distro and one Debian-derived distro. We could discuss the details in the PR.

**Comment by akash-akya** [comment]
Perfect! I'll create a PR once basic things are ready, we can discuss from there

**Comment by akash-akya** [comment]
Hi @jcupitt,

Sorry for the long pause, got caught up in day job.

I have a sample setup [here](https://github.com/akash-akya/libvips/commit/a47b2fad49fe3c84b9b4801bc51e183fa87f53dc). Which build and push image to GitHub container registry. The current Dockerfile is what you are calling as full-fat build I think. Dockerfile is currently bit ugly to keep container size small.

GH action: https://github.com/akash-akya/libvips/actions/runs/3256491283
Container: https://github.com/akash-akya/libvips/pkgs/container/libvips

I think we can have multiple variants like you mentioned and scheduled builds.

So what you think?

**Comment by aradalvand** [comment]
This would've been a great idea, @akash-akya @jcupitt Any updates guys?
